# About Team Genomics_Two_C
Hii all! Meet Team Genomics_Two_C ! We are a great team of 14 members from diverse background, but what draws us together is our common interest in genomics.

# HackBio Internship 2021 Stage_2 Task
Replicate a bioinformatic tutorial that relates to team biostack, i.e., genomics

## Project Title: Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data.
Tutorial Link: https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html 

### Project Description
Massive amount of sequencing data that are generated and analysed in real-time can be used to effectively monitor higly infectious disease such as the ongoing COVID-19 pandemic. This effective monitoring allows researchers to track the emergence and transmission of mutants and comprehend the evolutionary dynamics as well.
SARS-CoV-2 sequence data is primarily generated using two sequencing platforms - Illumina and Oxford Nanopore, and various established library preparation procedures like Ampliconic and Metatranscriptomic. However, these data has no meaning in itself, unless it is analysed.
Thus, the Galaxy community will be used because it has high-quality analysis workflows to support sensitive identification of SARS-CoV-2 allelic variants (AVs) starting with allele frequencies as low as 5% from deep sequencing reads, generate user-friendly reports for batches of results and generate reliable and configurable consensus genome from called variants.

### Aim
The aim of this research is to extract annotated allelic variants in SARS-Cov-2 sequences and identify SARS-CoV-2 lineages using different tools and workflows in Galaxy. Team Genomics_two_c will be extracting annotated allergic variants in SARS-Cov-2 sequences to identify SARS-Cov-2 lineages using different tools and workflows in Galaxy.

### Workflow
Step1

### Contribution
|01|@Sadaf|Conceptualization of workflow|
|02|@Subhodeep, @Borode|Flyer Design|
|03|@Eslam55520, @poornima|Sample and Auxillary data retrieval|
|04|@poornima, @Sadaf|FastQC to annotated allelic variants|
|05|@Sadaf, @chimere, @Nandy| Annotated AVs per sample to AV summary|
|06|@Borode, @Rhoda, @Sadaf|AVs to consensus sequences|
|07|@Fatima, @Mohamed, @Nuiter|Consensus sequences to clade assignments with Pangolin|
|08|@Emma, @Subhodeep|Consensus sequences to clade assignments with Nextclade|
|09|@Preeti|Comparing Pangolin and Nextclade|
|10|@Iffy, @Sadaf, @Preeti|Create markdown on Github|
